cmd/thanos/flags.go

```diff
@@ -73,3 +75,99 @@ func regSelectorRelabelFlags(cmd *kingpin.CmdClause) *extflag.PathOrContent {
 		false,
 	)
 }
+
+func parseFlagsToGrpcConfig(cmd *kingpin.CmdClause) grpcConfig {
+	return grpcConfig{
+		bindAddress: *cmd.Flag("grpc-address",
+			"Listen ip:port address for gRPC endpoints (StoreAPI). Make sure this address is routable from other components.").
+			Default("0.0.0.0:10901").String(),
+		gracePeriod: *modelDuration(cmd.Flag("grpc-grace-period",
+			"Time to wait after an interrupt received for GRPC Server.").
+			Default("2m")),
+		tlsSrvCert: *cmd.Flag("grpc-server-tls-cert",
+			"TLS Certificate for gRPC server, leave blank to disable TLS").
+			Default("").String(),
+		tlsSrvKey: *cmd.Flag("grpc-server-tls-key",
+			"TLS Key for the gRPC server, leave blank to disable TLS").
+			Default("").String(),
+		tlsSrvClientCA: *cmd.Flag("grpc-server-tls-client-ca",
+			"TLS CA to verify clients against. If no client CA is specified, there is no client verification on server side. (tls.NoClientCert)").
+			Default("").String(),
+	}
+}
+
+func parseFlagsToHttpConfig(cmd *kingpin.CmdClause) httpConfig {
+	return httpConfig{
+		bindAddress: *cmd.Flag("http-address",
+			"Listen host:port for HTTP endpoints.").
+			Default("0.0.0.0:10902").String(),
+		gracePeriod: *modelDuration(cmd.Flag("http-grace-period",
+			"Time to wait after an interrupt received for HTTP Server.").
+			Default("2m")),
+	}
+}
+
+func parseFlagsToPrometheusConfig(cmd *kingpin.CmdClause) prometheusConfig {
+	return prometheusConfig{
+		url: *cmd.Flag("prometheus.url",
+			"URL at which to reach Prometheus's API. For better performance use local network.").
+			Default("http://localhost:9090").URL(),
+		readyTimeout: *cmd.Flag("prometheus.ready_timeout",
+			"Maximum time to wait for the Prometheus instance to start up").
+			Default("10m").Duration(),
+	}
+}
+
+func parseFlagsToConnectionConfig(cmd *kingpin.CmdClause) connectionConfig {
+	return connectionConfig{
+		maxIdleConns: *cmd.Flag("receive.connection-pool-size",
+			"Controls the http MaxIdleConns. Default is 0, which is unlimited").
+			Int(),
+		maxIdleConnsPerHost: *cmd.Flag("receive.connection-pool-size-per-host",
+			"Controls the http MaxIdleConnsPerHost").
+			Default("100").Int(),
+	}
+}
+
+func parseFlagsToTsdbConfig(cmd *kingpin.CmdClause) tsdbConfig {
+	return tsdbConfig{
+		path: *cmd.Flag("tsdb.path",
+			"Data directory of TSDB.").
+			Default("./data").String(),
+	}
+}
+
+func parseFlagsToReloaderConfig(cmd *kingpin.CmdClause) reloaderConfig {
+	return reloaderConfig{
+		confFile: *cmd.Flag("reloader.config-file",
+			"Config file watched by the reloader.").
+			Default("").String(),
+		envVarConfFile: *cmd.Flag("reloader.config-envsubst-file",
+			"Output file for environment variable substituted config file.").
+			Default("").String(),
+		ruleDirectories: *cmd.Flag("reloader.rule-dir",
+			"Rule directories for the reloader to refresh (repeated field).").
+			Strings(),
+	}
+}
+
+func parseFlagsToShipperConf(cmd *kingpin.CmdClause) shipperConfig {
```

Comment:
We can follow Cortex way of doing thing and have this as just `RegisterFlag` method on shipperConfig struct directly.


```suggestion
func (c *shipperConfig) RegisterFlag(cmd *kingpin.CmdClause) {
```